sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -125,11 +126,7 @@ case class CheckOverflow(
       queryContext)
 
   override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
-    val errorContextCode = if (nullOnOverflow) {
-      "\"\""
-    } else {
-      ctx.addReferenceObj("errCtx", queryContext)
-    }
+    val errorContextCode = ctx.addReferenceObj("errCtx", queryContext)
```

Comment:
Will the original code work? It avoids adding a reference object.

Reply:
> It avoids adding a reference object.

Does it have real benefits? I just see overcomplicated code.

Reply:
Each object added will be serialized to executors. There should be benefit if a query contains many such expressions.

Reply:
ok. Reverted that back.